docs/source/traits_user_manual/advanced.rst

```diff
@@ -31,16 +31,20 @@ Dynamic Initialization
 When you define trait attributes using predefined traits, the Trait() factory
 function or trait handlers, you typically specify their default values
 statically. You can also define a method that dynamically initializes a trait
-attribute the first time that the attribute value is accessed. To do this, you
-define a method on the same class as the trait attribute, with a name based on
-the name of the trait attribute:
+attribute. To do this, you define a method on the same class as the trait
+attribute, with a name based on the name of the trait attribute:
 
 .. index:: default value; method
 
 .. method:: _name_default()
 
 This method initializes the *name* trait attribute, returning its initial value.
-The method overrides any default value specified in the trait definition.
+The method overrides any default value specified in the trait definition. The
+initializer is called when:
+
+1. the attribute value is accessed the first time or
+2. an instance is constructed with a specific value, if there is a change
+   handler defined for the trait (see :ref:`static-notification`).
```

Comment:
You might add something about the "why" here: it's needed to provide the "old" value to the trait notification system.